src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -406,7 +421,8 @@ synchronized void flush() throws InterruptedException, IOException, DeltaFailure
   }
 
   private void mergeTableChanges(TableBlob blob) throws DeltaFailureException, InterruptedException {
-    TableId stagingTableId = TableId.of(project, blob.getDataset(), stagingTablePrefix + blob.getTable());
+    String normalizedStagingTableName = normalize(stagingTablePrefix + blob.getTable());
```

Comment:
blob.getTable() should already be normalized right? Is this just to normalize the table prefix? If so, we can normalize it once in the constructor rather than every time here.

Reply:
blob.getTable() will be already normalized. There is an edge case that, if the normalized table name has length around 1020 and the normalized staging prefix is 8 for example. Then if we just combine them, it will have 1028 length which will exceed the 1024 limit. That is reason I initially do the normalization here.

Reply:
I see, makes sense

Reply:
Cool. I have updated the PR with new builder().